src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt

```diff
@@ -14,8 +14,11 @@ import java.io.File
 
 data class RustToolchain(val location: String) {
 
+    private fun hasExecuteable(exec: String): Boolean =
```

Comment:
Yes, this is exactly what I had in mind! 

And now that this is written this way, I see that the old code has probably had a bug, because it did check for the presence of `cargo` and not `rustc` in `queryRustcVersion`!